contributors/devel/sig-testing/flaky-tests.md

```diff
@@ -140,34 +158,242 @@ release-blocking flakes. Therefore we have the following guidelines:
+can take extra effort. Try following these steps:
+
+1. Run each test in the package individually to figure out the average runtime.
+2. Stress each test individually, bounding the timeout to 100 times the average run time.
+3. Isolate the particular test that is deadlocking.
+4. Add debug output to figure out what is causing the deadlock.
+
+Hopefully this can help narrow down exactly where the deadlock is occurring,
+revealing a simple fix!
+
+## Deflaking e2e tests
+
+A flaky [end-to-end (e2e) test](e2e-tests-kubetest2.md) offers its own set of
+challenges. In particular, these tests are difficult because they test the
+entire Kubernetes system. This can be both good and bad. It can be good because
+we want the entire system to work when testing, but an e2e test can also fail
+because of something completely unrelated, such as failing infrastructure or
+misconfigured volumes. Be aware that you can't simply look at the title of an
+e2e test to understand exactly what is being tested. If possible, look for unit
+and integration tests related to the problem you are trying to solve.
+
+### Gathering information
+
+The first step in deflaking an e2e test is to gather information. We capture a
+lot of information from e2e test runs, and you can use these artifacts to gather
+information as to why a test is failing.
+
+Use the [Prow Status](https://prow.k8s.io/) tool to collect information on
+specific test jobs. Drill down into a job and use the **Artifacts** tab to
+collect information. For example, with
+[this particular test job](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/directory/pull-kubernetes-e2e-gce/1296558932902285312),
+we can collect the following:
+
+* `build-log.txt`
+* In the control plane directory: `artifacts/e2e-171671cb3f-674b9-master/`
+  * `kube-apiserver-audit.log` (and rotated files)
+  * `kube-apiserver.log`
+  * `kube-controller-manager.log`
+  * `kube-scheduler.log`
+  * And more!
+
+The `artifacts/` directory will contain much more information. From inside the
+directories for each node:
+- `e2e-171671cb3f-674b9-minion-group-drkr`
+- `e2e-171671cb3f-674b9-minion-group-lr2z`
+- `e2e-171671cb3f-674b9-minion-group-qkkz`
+
+Look for these files:
+* `kubelet.log`
+* `docker.log`
+* `kube-proxy.log`
+* And so forth.
+
+### Filtering and correlating information
+
+Once you have gathered your information, the next step is to filter and
+correlate the information. This can require some familiarity with the issue you are tracking
+down, but look first at the relevant components, such as the test log, logs for the API
+server, controller manager, and `kubelet`.
+
+Filter the logs to find events that happened around the time of the failure and
+events that occurred in related namespaces and objects.
+
+The goal is to collate log entries from all of these different files so you can
+get a picture of what was happening in the distributed system. This will help
+you figure out exactly where the e2e test is failing.
```

Comment:
FYI @brianpursley
```suggestion
The goal is to collate log entries from all of these different files so you can
get a picture of what was happening in the distributed system. This will help
you figure out exactly where the e2e test is failing. One tool that may help you
with this is [k8s-e2e-log-combiner](https://github.com/brianpursley/k8s-e2e-log-combiner)
```

Reply:
/cc @brianpursley 
(should have used this the first time)